### ISSUE TITLE
Rubocop 0.91 failure: undefined method `<<' for #<Truffle::Digest:0x2aaf8>

### ISSUE THREAD

**Issue by gogainda** [oracle/truffleruby#2100]
```
0 files inspected, no offenses detected

undefined method `<<' for #<Truffle::Digest:0x2aaf8>

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/result_cache.rb:183:in `rubocop_checksum'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/result_cache.rb:94:in `initialize'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/runner.rb:135:in `new'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/runner.rb:135:in `cached_result'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/runner.rb:140:in `file_offense_cache'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/runner.rb:127:in `file_offenses'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/runner.rb:118:in `process_file'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/runner.rb:97:in `block in each_inspected_file'

<internal:core> core/enumerable.rb:448:in `inject'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/runner.rb:96:in `each_inspected_file'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/runner.rb:82:in `inspect_files'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/runner.rb:43:in `run'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/cli/command/execute_runner.rb:25:in `execute_runner'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/cli/command/execute_runner.rb:17:in `run'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/cli/command.rb:11:in `run'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/cli/environment.rb:18:in `run'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/cli.rb:65:in `run_command'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/cli.rb:72:in `execute_runners'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/lib/rubocop/cli.rb:41:in `run'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/exe/rubocop:13:in `block in <top (required)>'

/home/travis/.rvm/rubies/truffleruby-head/lib/mri/benchmark.rb:308:in `realtime'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/gems/rubocop-0.91.0/exe/rubocop:12:in `<top (required)>'

<internal:core> core/kernel.rb:402:in `load'

<internal:core> core/kernel.rb:402:in `load'

/home/travis/build/athityakumar/colorls/vendor/bundle/truffleruby/20.3.0-dev-983a39e6/bin/rubocop:23:in `<top (required)>'

<internal:core> core/kernel.rb:402:in `load'

<internal:core> core/kernel.rb:402:in `load'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/cli/exec.rb:63:in `kernel_load'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/cli/exec.rb:28:in `run'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/cli.rb:476:in `exec'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor/command.rb:27:in `run'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor/invocation.rb:127:in `invoke_command'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor.rb:399:in `dispatch'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/cli.rb:30:in `dispatch'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/vendor/thor/lib/thor/base.rb:476:in `start'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/cli.rb:24:in `start'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/exe/bundle:46:in `block in <top (required)>'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/lib/bundler/friendly_errors.rb:123:in `with_friendly_errors'

/home/travis/.rvm/gems/truffleruby-head/gems/bundler-2.1.4/exe/bundle:34:in `<top (required)>'

<internal:core> core/kernel.rb:402:in `load'

<internal:core> core/kernel.rb:402:in `load'

/home/travis/.rvm/gems/truffleruby-head/bin/bundle:23:in `<main>'

The command "bundle exec rubocop" exited with 2.
```
from [here](https://travis-ci.org/github/athityakumar/colorls/builds/727860910)

**Comment by chrisseaton** [comment]
Looks like `Digest` is still somewhat under specified https://github.com/ruby/spec/tree/master/library/digest.

**Comment by eregon** [comment]
From https://github.com/rubocop-hq/rubocop/blob/e0fc1c0437bcea8c610c850dc06e59ed36e0c9c8/lib/rubocop/result_cache.rb#L183
Looks like we're just missing `Digest#<<`, which seems an alias of `update`.
@bjfish Can you look into that?

Maybe we can add some methods check for Digest as it's both badly documented and tested.
Adding `Digest Digest::Base Digest::Class Digest::Instance Digest::MD5 Digest::SHA1` to https://github.com/oracle/truffleruby/blob/master/spec/truffle/methods_spec.rb would be simplest if that works.

**Comment by chrisseaton** [comment]
It's also confusing with `Base` and `Instance`. If you look in `Digest` you don't see these things.

**Comment by eregon** [comment]
Fixed in aa495587766ec1f21ee8c98a5d04ea3739836a1b by @bjfish.
